Lombiq.NodeJs.Extensions/scripts/handle-error.js

```diff
@@ -3,7 +3,15 @@
  */
 
 const os = require('os');
-const chalk = require('chalk');
+
+// Treat this dependency as optional because it's not available everywhere.
```

Comment:
> it's not available everywhere

What exactly does that mean?

Reply:
It wasn't available when executing from the solution root via _CreateGlobalConfigFiles.targets_. I'm not sure why. This is the sole external dependency of the script and it's only cosmetic, so in principle it's nicer to make it optional.

But I don't understand why we need this package at all. MSBuild should color and format the output by itself when the terminal supports it. Also on GHA the MSBuild error/warning is captured and transformed into an annotation.

Reply:
This is for coloring output in the VS Task Explorer Runner, which didn't show any colors otherwise.

Can you point me to the erroneous run? Windows or Linux? I'm asking because I haven't had any problems with this in the past. (Maybe, I didn't have any errors to report, either, though.)

Reply:
I had it on Linux, but that's probably because [the error](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/4143907294/jobs/7166322503) only appeared on Linux as it was related to the path used ([see commit](https://github.com/Lombiq/NodeJs-Extensions/pull/24/commits/f25c11cab070b60d804f081804e05d75cfce91c2)).

Reply:
I'm hesitant to keeping the change, because I don't see any trace of `chalk` causing any problems looking at the very detailed binary log:
![image](https://user-images.githubusercontent.com/164360/218479469-afd6c79c-1b9f-4464-b04f-c7b3147d2363.png)

There's not a single mention of `chalk` in the whole log.

Reply:
Dunno, it was days ago. But it's pretty easy to reproduce it:
![image](https://user-images.githubusercontent.com/4007293/218481869-07e2747c-9fd3-43e0-94ed-bbf76fcd6b4e.png)
If you don't know a quick fix for this, then it would be better to open a new issue as it shouldn't be a blocker.

Also note that I will absolutely use _handle-error.js_ wherever it makes sense, not just inside NE scripts (e.g. it's in LV too). I'd rather it not have required dependencies in error handling because that could break things silently.

Reply:
I've run `dotnet build` after replacing `pnpm add` with `pnpm addy`, and did not see any problem with chalk missing:

![image](https://user-images.githubusercontent.com/164360/218483724-c7fa54b8-5684-4208-a05a-9208d2833bc1.png)

So I feel, that the change in code here is unnecessary.

Reply:
I can agree that the dependency might not be necessary for the core _handle-error_ script, so I suggest to move the coloring of the error into an optional callback parameter.

But in a new issue.

Reply:
Here it is: https://github.com/Lombiq/NodeJs-Extensions/issues/30.